airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -421,6 +421,26 @@ public List<CheckedConsumer<JdbcDatabase, Exception>> getCheckOperations(final J
     final List<CheckedConsumer<JdbcDatabase, Exception>> checkOperations = new ArrayList<>(
         super.getCheckOperations(config));
 
+    // if UI has schemas specified, check if the user has select access to any table
+    if (schemas != null && !schemas.isEmpty()) {
+      checkOperations.add(database -> {
+        for (final String schema : schemas) {
+          LOGGER.debug("Checking if the user can perform select to any table in schema: {}", schema);
+          ResultSet tablesOfSchema = database.getMetaData().getTables(null, null, "%", null);
```

Comment:
Shouldn't `schema` be passed as the second argument here? Also, perhaps you can pass `null` instead of `"%"` (not sure)

Reply:
yes!

Reply:
TIL! thanks